common/lib/common-utils/.eslintrc.js

```diff
@@ -5,16 +5,55 @@
 
 module.exports = {
     "extends": [
-        "@fluidframework/eslint-config-fluid"
+        "@fluidframework/eslint-config-fluid/recommended"
     ],
     "parserOptions": {
-        "project": [ "./tsconfig.json", "./src/test/mocha/tsconfig.json", "./src/test/jest/tsconfig.json", "./src/test/types/tsconfig.json" ]
+        "project": [
+            "./tsconfig.json",
+            "./src/test/mocha/tsconfig.json",
+            "./src/test/jest/tsconfig.json",
+            "./src/test/types/tsconfig.json",
+        ]
     },
     "rules": {
         "@typescript-eslint/no-non-null-assertion": "off",
         "@typescript-eslint/strict-boolean-expressions": "off",
+        "@typescript-eslint/no-implied-eval": "error",
+        "@typescript-eslint/promise-function-async": "error",
+        "default-case": "error",
         "no-bitwise": "off",
         "no-null/no-null": "off",
-        "prefer-rest-params": "off"
-    }
+        "prefer-rest-params": "off",
+        "import/no-internal-modules": "error",
+        "import/no-nodejs-modules": [
+            "error", {
+                "allow": [
+                    "node:assert",
+                    "node:events",
+                    "node:fs",
+                    "node:http",
+                    "node:path",
+                    "node:perf_hooks",
+                    "node:process",
+                ]
+            }
+        ],
+    },
+    "overrides": [
+        {
+            // Rules only for test files
+            "files": [
+                "**/*.spec.ts",
+                "src/test/**"
+            ],
+            "rules": {
+                "@typescript-eslint/explicit-function-return-type": "off",
```

Comment:
This is already off from eslint7.js

I'm also not sure how I feel about this rule.  Assuming we have noImplicitAny and no-unsafe-return enabled to protect against the more-dangerous scenarios, I kind of like the ease of inferred typing.  Not a very strong opinion though.

Reply:
Explicit typing is helpful with api-extractor, because it prevents e.g. "flip-flopping" of union types when they're inferred. [This extension](https://marketplace.visualstudio.com/items?itemName=ebrithil30.vscode-ts-auto-return-type) for VS Code makes cleaning up existing code pretty easy.

Reply:
I'm on the fence here too, but if it fixes the api-extractor problem then that's really good.  We should use option `allowExpressions: true`, I think that will cover a lot of cases where this would be onerous and redundant with inferred typing.  That's my hunch anyway.